### DISCUSSION TITLE
Trigger : Stack Depth Limit reached

### DISCUSSION THREAD

**Discussion by ijaspreetsinghh** [supabase/supabase#6821] [Questions] [answered]
Im trying to write a trigger that will calculate a value using select and update on a specific row, whenever there is an update/insert operation. on trigger getting error 'Stack depth limit reached'.

    CREATE OR REPLACE FUNCTION update_company_qty_for_insert_update()
    RETURNS TRIGGER AS
    $$
    BEGIN
    update "InventoryItems" set company_qty = (SELECT  sum("InventoryItems".quantity_on_hand )  from "InventoryItems"  where 
    "InventoryItems".upc=NEW.upc  and "InventoryItems".company_id=NEW.company_id) where   "InventoryItems".upc=NEW.upc  and 
    "InventoryItems".company_id=NEW.company_id;
    RETURN NEW;
    END
    $$ LANGUAGE plpgsql;

**Comment by GaryAustin1** [comment, marked as answer]
Is this an update trigger on InventoryItems?  If so you are doing an update which will call the trigger again.   If that is the case then don’t do an update in the function  just do a before trigger and and set new.company_qty to the calculated value. 

Edit, in relooking at your code, you might be trying to update several rows.  Then my above suggestion won't work.  And you probably need to do this update thru a function that updates/inserts the new row and also updates the column in all the rows needed.  By doing it all in a single function then you avoid the need for trigger on update which won't work for updating the same table.

Some other info: https://stackoverflow.com/questions/708562/prevent-recursive-trigger-in-postgresql

**Reply by ijaspreetsinghh** [reply]
Yes i was trying to update multiple rows,
thanks a lot, that pg_trigger_depth() < 1 worked for me.
